### PR TITLE
implement some traits on FormatContext

### PR DESCRIPTION
**Pull request by MarcAntoine-Arnaud**

### CODE REVIEW

**Comment by MarcAntoine-Arnaud** [comment]
@Romanelaf can you take a look on it ? and publish a new version too ?

File: src/probe/deep.rs [hunk 1]
```diff
@@ -178,10 +176,9 @@ fn deep_probe_mxf_sample() {
...
-  let mut file = File::open("tests/deep_probe.json").unwrap();
-  let mut contents = String::new();
-  file.read_to_string(&mut contents).unwrap();
+  // println!("{}", serde_json::to_string(&probe).unwrap());
 
-  let reference: DeepProbe = serde_json::from_str(&contents).unwrap();
-  assert_eq!(probe, reference);
+  // let content = std::fs::read_to_string("tests/deep_probe.json").unwrap();
+  // let reference: DeepProbe = serde_json::from_str(&content).unwrap();
+  // assert_eq!(probe, reference);
```
Comment (Romanelaf) [inline src/probe/deep.rs:183]: I don't understand why this test doesn't pass, it works locally

Comment (Romanelaf) [reply src/probe/deep.rs:183]: It's the new ffmpeg version 4.3.1 (released on 2020-07-11). They've changed the duration type of silence filter

Comment (Romanelaf) [reply src/probe/deep.rs:183]: I will update deep probe in another pull request

Comment (MarcAntoine-Arnaud) [reply src/probe/deep.rs:183]: Thank you to spent time to check that !

**Pull request merged**